packages/client/hmi-client/src/views/DataExplorer.vue

```diff
@@ -345,12 +345,19 @@ const executeSearch = async () => {
 			}
 		}
 	});
-	const modelSearchParams = searchParamsWithFacetFilters?.[ResourceType.MODEL] || {
-		filters: clientFilters.value
-	};
-	const datasetSearchParams = searchParamsWithFacetFilters?.[ResourceType.DATASET] || {
-		filters: clientFilters.value
-	};
+
+	let modelSearchParams;
```

Comment:
Can we use the previous terser syntax here?

Reply:
The previous syntax is what was breaking the use of facets previously.
It looks cleaner but is not correct

What is says is: if searchParamsWithFacetFilters?.[ResourceType.DATASET]
What we want is: if searchParamsWithFacetFilters?.[ResourceType.DATASET]?.filters

If the if/else statements are too ugly I am happy to try a cleaner approach. Let me know if you have suggestions!

Reply:
Does 
```
const modelSearchParams = searchParamsWithFacetFilters?.[ResourceType.DATASET]?.filters || {
		filters: clientFilters.value
	};
``` work?

Reply:
No, then we are assigning the type Filters to the type SearchParameters.

Reply:
Ah okay, so the two different possible values have different types?